truffle/com.oracle.truffle.api/src/com/oracle/truffle/api/frame/FrameInstance.java

```diff
@@ -47,9 +48,18 @@
     /** @since 0.8 or earlier */
     boolean isVirtualFrame();
 
-    /** @since 0.8 or earlier */
+    /**
+     * @return the node that is calling the next {@link CallTarget target} or <code>null</code> if
+     *         its the topmost frame or the next {@link CallTarget target} was called without a
+     *         {@link TruffleRuntime#createDirectCallNode(CallTarget)} or
+     *         {@link TruffleRuntime#createIndirectCallNode() indirect} call node.
+     * @since 0.8 or earlier
+     **/
```

Comment:
Better?
```
    /**
     * @return the node that represents the caller of the next upper {@link CallTarget target} in
     *         the stack trace. Null in case there is no upper target or if the target was not
     *         invoked using a {@link TruffleRuntime#createDirectCallNode(CallTarget) direct} or
     *         {@link TruffleRuntime#createDirectCallNode(CallTarget) indirect} call node.
     *
     * @since 0.8 or earlier
     **/
    Node getCallNode();
```